flytekit/core/type_engine.py

```diff
@@ -407,17 +407,18 @@ def literal_map_to_kwargs(
         return {k: TypeEngine.to_python_value(ctx, lm.literals[k], v) for k, v in python_types.items()}
 
     @classmethod
-    def dict_to_literal_map(cls, ctx: FlyteContext, d: typing.Dict[str, typing.Any]) -> LiteralMap:
+    def dict_to_literal_map(cls, ctx: FlyteContext, d: typing.Dict[str, typing.Any], m: typing.Dict[str, type]) -> LiteralMap:
```

Comment:
Let's make the new argument optional - this is presumably a public API (though I doubt very many people use it), so we should keep the api backwards compatible.